Xamarin.Forms.Core/BindableProxy.cs

```diff
@@ -0,0 +1,336 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Reflection;
+
+namespace Xamarin.Forms
+{
+	class BindableProxy : BindableObject, IDisposable
+	{
+		readonly object _targetObject;
+		readonly string _targetProperty;
+		readonly string _targetEventName;
+		readonly PropertyInfo _propInfo;
+		List<MethodInfo> _setMethodsInfo;
+		List<MethodInfo> _getMethodsInfo;
+		List<Type> _possibleParameterTypes;
+		INativeValueConverter nativeValueConverter;
+		Action<object, object> _callbackSetValue;
+		Func<object> _callbackGetValue;
+		bool _isSettingFromProxy;
+
+		NativeViewEventListener _eventListener;
+		bool _disposed;
```

Comment:
The number of instance variable looks like an indicator that something is not right